configuration/src/main/java/io/camunda/configuration/S3.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.configuration;
+
+import io.camunda.configuration.UnifiedConfigurationHelper.BackwardsCompatibilityMode;
+import java.time.Duration;
+import java.util.Set;
+
+public class S3 {
+  private static final String PREFIX = "camunda.data.backup.s3.";
+  private static final Set<String> LEGACY_BUCKETNAME_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.bucketName");
+  private static final Set<String> LEGACY_ENDPOINT_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.endpoint");
+  private static final Set<String> LEGACY_REGION_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.region");
+  private static final Set<String> LEGACY_ACCESSKEY_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.accessKey");
+  private static final Set<String> LEGACY_SECRETKEY_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.secretKey");
+  private static final Set<String> LEGACY_APICALLTIMEOUT_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.apiCallTimeout");
+  private static final Set<String> LEGACY_FORCEPATHATYLEACCESS_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.forcePathStyleAccess");
+  private static final Set<String> LEGACY_COMPRESSION_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.compression");
+  private static final Set<String> LEGACY_MAXCONCURRENTCONNECTIONS_PROPERTIES =
+      Set.of("zeebe.broker.data.backup.s3.maxConcurrentConnections");
+  private static final Set<String> LEGACY_CONNECTIONAQUISITIONTIMEOUT_PROPERTIES =
```

Comment:
nit: consider adding delimiters for readability, e.g.
```suggestion
  private static final Set<String> LEGACY_CONNECTION_AQUISITION_TIMEOUT_PROPERTIES =
```